lib/rucio/daemons/tracer/kronos.py

```diff
@@ -389,7 +389,8 @@ def run_once_kronos_file(heartbeat_handler: HeartbeatHandler, return_values: Dic
             conn.subscribe(destination=config_get('tracer-kronos', 'queue'), ack='client-individual', id=subscription_id, headers={'activemq.prefetchSize': prefetch_size})
 
 
-def kronos_dataset(once: bool = False, dataset_queue: Queue = None, sleep_time: int = 60):
+def kronos_dataset(dataset_queue: Queue, once: bool = False, sleep_time: int = 60):
+    return_values = {'heartbeat_handler': HeartbeatHandler("kronos-dataset", 10)}
```

Comment:
This is so dirty that it hurts :D

Reply:
It really does man, but following the discussions above I have limited options. 
I propose that we can worry about reducing the size of the unhandled AMQ messages later. In the future, we can maybe tweak what the daemons should do before exiting. For eg, we could create a new callback method to the run_daemon method for this. 
If we agree for now, then I can happily get rid of the custom HeartbeatHandler and return_values dict.
What do you think?

Reply:
Sorry that I wasn't clear enough. I did approve the PR, so it's good for me to merge it this way. It just hurts to see it :grinning: